src/main/scala/mesosphere/marathon/core/task/jobs/impl/ExpungeOverdueLostTasksActor.scala

```diff
@@ -48,16 +49,13 @@ class ExpungeOverdueLostTasksActor(
   }
 
   def filterLostGCTasks(tasks: Map[PathId, AppTasks]): Iterable[Task] = {
-    def isTimedOut(task: Task): Boolean = {
-      task.mesosStatus.fold(false) { status =>
+    def isTimedOut(taskStatus: Option[TaskStatus]): Boolean = {
+      taskStatus.fold(false) { status =>
         val age = clock.now().toDateTime.minus(status.getTimestamp.toLong * 1000).getMillis.millis
         age > config.taskLostExpungeGC
       }
     }
-    tasks.values.flatMap(_.tasks.filter {
-      case TemporarilyUnreachable(task) if isTimedOut(task) => true
-      case _ => false
-    })
+    tasks.values.flatMap(_.tasks.filter(task => isTimedOut(task.mesosStatus)))
```

Comment:
This is the place where we actually only want to have the MarathonTaskStatus and the timestamp of when the task entered that state. Since the Task only offers `stagedAt` and `startedAt`, we need to rely on the mesos TaskStatus here, which is quite unfortunate.

In the _hopefully near_ future, this should be something like
`isTimedOut(task.status.since)`

I don't think this improvement should/can be part of this PR though.